packages/calcite-components/src/components/table/table.tsx

```diff
@@ -343,7 +347,8 @@ export class Table extends LitElement {
     });
   }
 
-  private updateSelectedItems(emit?: boolean): void {
+  private async updateSelectedItems(emit?: boolean): Promise<void> {
+    await this.updateComplete;
```

Comment:
Can you double check that all calls to `updateComplete` are necessary and don’t introduce unnecessary delays?

Reply:
good catch, this isn't required.